presto-main/src/main/java/com/facebook/presto/util/JsonUtil.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.util;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Throwables;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.io.Writer;
+
+import static java.util.Objects.requireNonNull;
+
+public class JsonUtil
+{
+    private JsonUtil()
+    {
+    }
+
+    public static String toJsonWithLengthLimit(ObjectMapper objectMapper, Object value, int lengthLimit)
+    {
+        try (StringWriter stringWriter = new StringWriter();
+                LengthLimitedWriter lengthLimitedWriter = new LengthLimitedWriter(stringWriter, lengthLimit)) {
+            objectMapper.writeValue(lengthLimitedWriter, value);
+            return stringWriter.getBuffer().toString();
+        }
+        catch (LengthLimitedWriter.LengthLimitExceededException e) {
+            return "{\"discarded\":\"length limit exceeded\"}";
```

Comment:
I think we should just leave it out. This will make it harder to process the json structures because the won't have the same shape at the top level